pandas/core/groupby/ops.py

```diff
@@ -663,6 +663,14 @@ def _aggregate_series_pure_python(self, obj: Series, func):
             counts[label] = group.shape[0]
             result[label] = res
 
+            if isinstance(res, np.bool_):
+                if is_extension_array_dtype(group.dtype):
+                    # FIXME: kludge for DecimalArray on numpy<1.15 returning
+                    #  np.bool_(False) when we have all-NA DecimalArray
+                    if group.isna().all():
+                        res = group.dtype.na_value
+                        result[label] = res
```

Comment:
It's not clear to why this is the correct fix. You are changing the result of the function, without necessarily knowing what the function is?

Reply:
yah, i want to take another shot at fixing this behavior inside `DecimalArray.__array_whatever__`

Reply:
tentatively looks like a fix in `DecimalArray._reduce` gets the job done, so the change here has been reverted.  We'll see if the CI agrees.